opencti-platform/opencti-graphql/src/database/middleware.js

```diff
@@ -1987,7 +1987,30 @@ export const updateAttribute = async (context, user, id, type, inputs, opts = {}
   if (!initial) {
     throw FunctionalError('Cant find element to update', { id, type });
   }
-  return updateAttributeFromLoadedWithRefs(context, user, initial, inputs, opts);
+
+  // region confidence control
+  if (!user.effective_confidence_level) {
+    throw FunctionalError('User has no effective max confidence level and cannot update the element', { id });
+  }
+  const elementConfidence = Math.max(Math.min(initial.confidence ?? 0, 100), 0);
+  const userMaxConfidenceLevel = user.effective_confidence_level.max_confidence;
+  if (userMaxConfidenceLevel < elementConfidence) {
+    throw FunctionalError('User effective max confidence level is insufficient to update this element', { id });
+  }
+  // if the attribute updated is "confidence", new value cannot exceed user's max confidence
+  const newInputs = inputs.map((input) => {
```

Comment:
Maybe throw could be a better solution

Reply:
Yes, that would be more consistent behavior.

Reply:
after some more discussions, we want to be consistent with respect to create and upsert strategies, so we simply crop the input value.